lib/inferno/dsl/fhir_evaluation/rules/all_must_supports_present.rb

```diff
@@ -0,0 +1,329 @@
+            else
+              message = 'Found Profiles with not all MustSupports represented:'
+              missing_items_by_profile.each do |profile_url, missing_items|
+                message += "\n\t\t#{profile_url}: #{missing_items.join(', ')}"
+              end
+              result = EvaluationResult.new(message, rule: self)
+            end
+            context.add_result result
+          end
+
+          def pick_resources_for_profile(profile, context)
+            conformance_options = context.config.data['Rule']['AllMustSupportsPresent']['ConformanceOptions'].to_options
+
+            # Unless specifically looking for Bundles, break them out into the resources they include
+            all_resources =
+              if profile.type == 'Bundle'
+                context.data
+              else
+                flatten_bundles(context.data)
+              end
+
+            all_resources.filter do |r|
+              conforms_to_profile?(r, profile, conformance_options, context.validator)
+            end
+          end
+
+          # perform_must_support_test is invoked from DSL assertions, allows customizing the metdata with a block.
+          # Customizing the metadata may add, modify, or remove items.
+          # For instance, US Core 3.1.1 Patient "Previous Name" is defined as MS only in narrative.
+          # Choices are also defined only in narrative.
+          def perform_must_support_test(profile, resources, ig, requirement_extension = nil)
+            profile_metadata = extract_metadata(profile, ig, requirement_extension)
+            yield profile_metadata if block_given?
+
+            perform_must_support_test_with_metadata(resources, profile_metadata)
+          end
+
+          # perform_must_support_test_with_metadata is invoked from either option above,
+          #   with the metadata to be used as the basis for the check
+          # (or can be invoked directly from a test if you want to completely overwrite the metadata)
+          def perform_must_support_test_with_metadata(resources, profile_metadata)
+            return if resources.blank?
+
+            @metadata = profile_metadata
+
+            missing_elements(resources)
+            missing_slices(resources)
+            missing_extensions(resources)
+
+            handle_must_support_choices if metadata.must_supports[:choices].present?
+
+            missing_must_support_strings
+          end
+
+          def extract_metadata(profile, ig, requirement_extension = nil)
+            MustSupportMetadataExtractor.new(profile.snapshot.element, profile, profile.type, ig, requirement_extension)
+          end
+
+          def handle_must_support_choices
+            handle_must_support_element_choices
+            handle_must_support_extension_choices
+            handle_must_support_slice_choices
+          end
+
+          def handle_must_support_element_choices
+            missing_elements.delete_if do |element|
```

Comment:
I added a new choice type for US Core 8., which is called :elements. Each element choice has a path and a fixed_value. This is an expansion of :paths choice. You can see the code changes [here](https://github.com/inferno-framework/us-core-test-kit/pull/219/files#diff-827a7b2e0785340a5bcfcf8affe823fa5644fef636e130d5cb92b963afd7ba6a)